sdk/eventhub/azure-eventhubs/azure/eventhub/aio/producer_async.py

```diff
@@ -168,19 +160,22 @@ def _on_outcome(self, outcome, condition):
         :return: an EventDataBatch instance
         :rtype: ~azure.eventhub.EventDataBatch
         """
-        max_size = kwargs.get("max_size", None)
-        partition_key = kwargs.get("partition_key", None)
+
+        @_retry_decorator
```

Comment:
Why not just `await _retry_decorator(self.open)(timeout=...)`?

Reply:
fixed, it's neat now.